src/kv/kvtypes.h

```diff
@@ -116,6 +101,50 @@ namespace kv
     virtual crypto::Sha256Hash get_root() = 0;
   };
 
+  class Replicator
```

Comment:
As discussed, the `kv::Replicator` should only include the methods that the kv store requires.

Reply:
Do we need to keep a separate replicator if we have a generic consensus interface?

Reply:
Renamed Replicator to Consensus and for now keeping all the related information there. See #304 for future refactoring/improvements